lib/OperationHandler.cpp

```diff
@@ -64,87 +62,139 @@ void OperationHandler::SendGuestNotification(std::variant<DisconnectNotif, Signa
+        Log::Info(L"Notifying the client of a guest scan completion. Status: %ws", ToString(status));
 
-        if (m_clientCallbacks.OnGuestConnectRequestProgress)
+        if (m_clientObserver)
         {
-            m_clientCallbacks.OnGuestConnectRequestProgress(status);
+            m_clientObserver->OnGuestScanCompletion(status);
         }
     });
 }
 
 void OperationHandler::OnHostConnection(const GUID& interfaceGuid, const Ssid& ssid, DOT11_AUTH_ALGORITHM authAlgo)
 {
     // Always notify the client on a new host connection
-    m_clientNotificationQueue.Run([this, interfaceGuid, ssid, authAlgo] {
-        NotifyConnectionToClientSerialized(EventSource::Host, interfaceGuid, ssid, authAlgo);
+    m_clientNotificationQueue.Run([this, args = ProxyWifiObserver::ConnectCompleteArgs{interfaceGuid, ssid, authAlgo}] {
+        Log::Info(
+            L"Notifying the client of a host connection. Interface: %ws, Ssid: %ws, Auth Algo: %ws",
+            GuidToString(args.interfaceGuid).c_str(),
+            SsidToLogString({args.ssid.ucSSID, args.ssid.uSSIDLength}).c_str(),
+            Wlansvc::AuthAlgoToString(args.authAlgo).c_str());
+        if (m_clientObserver)
+        {
+            m_clientObserver->OnHostConnection(args);
+        }
     });
 }
 
 void OperationHandler::OnHostDisconnection(const GUID& interfaceGuid, const Ssid& ssid)
 {
     // Notify the client first
-    m_clientNotificationQueue.Run(
-        [this, interfaceGuid, ssid] { NotifyDisconnectionToClientSerialized(EventSource::Host, interfaceGuid, ssid); });
+    m_clientNotificationQueue.Run([this, args = ProxyWifiObserver::DisconnectCompleteArgs{interfaceGuid, ssid}] {
+        Log::Info(
+            L"Notifying the client of a host disconnection. Interface: %ws, Ssid: %ws",
+            GuidToString(args.interfaceGuid).c_str(),
+            SsidToLogString({args.ssid.ucSSID, args.ssid.uSSIDLength}).c_str());
+
+        if (m_clientObserver)
+        {
+            m_clientObserver->OnHostDisconnection(args);
+        }
+    });
 
     // If this is a spontaneous disconnection from the host on the interface currently used by the guest,
     // send a disconnect notification to the guest and mark it as disconnected
 
     // TODO guhetier: Remove this completely or keep it behind a flag for having it in the public lib?
 
     // m_serializedRunner.Run([this, interfaceGuid] {
-        // if (m_guestConnection && interfaceGuid == m_guestConnection->interfaceGuid)
-        // {
-        //     m_guestConnection.reset();
+    // if (m_guestConnection && interfaceGuid == m_guestConnection->interfaceGuid)
+    // {
+    //     m_guestConnection.reset();
 
-        //     Log::Trace(L"Sending Disconnection notification to the guest");
-        //     SendGuestNotification(DisconnectNotif{m_sessionId});
-        // }
+    //     Log::Trace(L"Sending Disconnection notification to the guest");
+    //     SendGuestNotification(DisconnectNotif{m_sessionId});
+    // }
```

Comment:
Is there still a reason to keep this around?

Reply:
I was thinking of adding an option to allow enabling it if wanted, since it could make sense depending how guest connection are driven. I didn't have time to do this yet.

Reply:
Sounds good!